polar-c-api/src/lib.rs

```diff
@@ -1,12 +1,49 @@
+// everything in this file is unsafe, clippy
+#![allow(clippy::not_unsafe_ptr_arg_deref)]
```

Comment:
Yeah. We can mark all the functions as unsafe. Just seems a little redundant since these are all extern anyway.

Reply:
These are all considered unsafe based on Rust's definition of unsafe (potential for undefined behavior) from what I can tell since they access potentially uninitialized memory.